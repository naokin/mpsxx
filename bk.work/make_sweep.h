#ifndef __MPSXX_MAKE_SWEEP_H
#define __MPSXX_MAKE_SWEEP_H

#include <vector>

#include "input.h"

namespace mpsxx {

/// Perform 1 fwd./bwd. sweep iteration
double make_sweep(const std::vector<double>& E, const DMRGInput& input, const size_t& iroot = 0);

} // namespace mpsxx

#endif // __MPSXX_MAKE_SWEEP_H
