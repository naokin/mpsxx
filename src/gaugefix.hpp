#ifndef __MPSXX_GAUGE_FIX_HPP
#define __MPSXX_GAUGE_FIX_HPP

#include <iostream>
#include <iomanip>

#include <btas/QSPARSE/QSTArray.h>

#include "input.h"
#include "canonicalize.hpp"
#include "compute_guesswave.hpp"
#include "fileio.h"

namespace mpsxx {

/// Compute expectation value <i|O|j>
template<class Q>
void gaugefix (const DMRGInput& input, const size_t& iroot = 0)
{
  using std::cout;
  using std::endl;
  using std::setw;

  size_t N = input.N_sites;
  size_t K = input.N_roots;

  assert(iroot < K);

  btas::QSTArray<double,3,Q> lmps;
  btas::QSTArray<double,3,Q> rmps;
  btas::QSTArray<double,3,Q> wave;

  if(0) { // algo 1

  load(wave,getfile("wave",input.prefix,0,iroot));
  for(size_t i = 0; i < N-1; ++i) {
    load(rmps,getfile("rmps",input.prefix,i+1,iroot));
    canonicalize(1,wave,lmps,rmps);
    save(lmps,getfile("lmps",input.prefix,i,iroot));
    wave = rmps;
  }
  for(size_t i = N-1; i > 0; --i) {
    btas::Normalize(wave);
    save(wave,getfile("wave",input.prefix,i,iroot));

    load(lmps,getfile("lmps",input.prefix,i-1,iroot));
    canonicalize(0,wave,lmps,rmps);
    save(rmps,getfile("rmps",input.prefix,i,iroot));
    wave = lmps;
  }
    btas::Normalize(wave);
    save(wave,getfile("wave",input.prefix,0,iroot));

  } else { // algo 2

  load(wave,getfile("wave",input.prefix,0,iroot));
  for(size_t i = 0; i < N-1; ++i) {
    load(lmps,getfile("lmps",input.prefix,i,  iroot));
    load(rmps,getfile("rmps",input.prefix,i+1,iroot));
    btas::QSTArray<double,3,Q> temp;
    compute_guesswave(1,lmps,wave,rmps,temp);
    wave = temp;
    save(wave,getfile("wave",input.prefix,i+1,iroot));
  }
  for(size_t i = N-1; i > 0; --i) {
    load(rmps,getfile("rmps",input.prefix,i,  iroot));
    load(lmps,getfile("lmps",input.prefix,i-1,iroot));
    btas::QSTArray<double,3,Q> temp;
    compute_guesswave(0,rmps,wave,lmps,temp);
    wave = temp;
    save(wave,getfile("wave",input.prefix,i-1,iroot));
  }

  } // end if
}

} // namespace mpsxx

#endif // __MPSXX_GAUGE_FIX_HPP
