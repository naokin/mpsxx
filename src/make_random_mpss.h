#ifndef __MPSXX_MPSGEN_MAKE_RANDOM_MPSS_H
#define __MPSXX_MPSGEN_MAKE_RANDOM_MPSS_H

#include <vector>
#include <btas/QSPARSE/QSTArray.h>

#include "input.h"
#include "fermion.h"

namespace mpsxx {

/// Generate quantum numbers for each boundary from MPO
///
/// \param sites quantum numbers for each site (physical index)
/// \param qt total quantum number of lattice
/// \param _max_quantum_blocks maximum number of quantum blocks on each boundary (0: no limitation by default)
std::vector<btas::Qshapes<fermion>>
generate_quantum_states (const std::vector<btas::Qshapes<fermion>>& sites, const fermion& qt, size_t max_quantum_blocks = 0);

void make_random_mpss (const DMRGInput& input, const size_t& iroot = 0);

}

#endif // __MPSXX_MPSGEN_MAKE_RANDOM_MPSS_H
