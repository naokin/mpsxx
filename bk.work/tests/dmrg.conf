
!restart

N    4
M   50

elec  4
spin  0

nroots 1

twosite
maxiter 200

tole 1.0e-6

