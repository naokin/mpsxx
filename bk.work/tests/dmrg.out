	****************************************************************************************************
				MPSXX::PROTOTYPE::DMRG::OPTIMIZATION 
	****************************************************************************************************
	====================================================================================================
			Generate & initialize MPS: number of sites =   4
	====================================================================================================
			Generating quantum states for each boundary 
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			Generating   MPS for site [   0 ] 
	----------------------------------------------------------------------------------------------------
			Generating   MPS for site [   1 ] 
	----------------------------------------------------------------------------------------------------
			Generating   MPS for site [   2 ] 
	----------------------------------------------------------------------------------------------------
			Generating   MPS for site [   3 ] 
	====================================================================================================
			Initializing MPS for site [   3 ] 
	----------------------------------------------------------------------------------------------------
			Initializing MPS for site [   2 ] 
	----------------------------------------------------------------------------------------------------
			Initializing MPS for site [   1 ] 
	----------------------------------------------------------------------------------------------------
			Initializing MPS for site [   0 ] 
	====================================================================================================
	====================================================================================================
		SWEEP ITERATION [    0 ] :: ROOT =  0
	====================================================================================================
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			FORWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   0 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::       0.9999999999999998
			micro iteration [  2 ] ::       0.5230547766287836
			micro iteration [  3 ] ::      -0.5565658942052466
			micro iteration [  4 ] ::      -1.8892801107919959
			micro iteration [  5 ] ::      -2.2371358163642201
			micro iteration [  6 ] ::      -2.3098375243117149
			micro iteration [  7 ] ::      -2.3511191541034151
			micro iteration [  8 ] ::      -2.3649844335184587
			micro iteration [  9 ] ::      -2.3657222276988223
			micro iteration [ 10 ] ::      -2.3658087893312416
			--------------------------------------------------
			macro iteration [  1 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -2.3658087893312434
			micro iteration [  2 ] ::      -2.3658241434312526
			micro iteration [  3 ] ::      -2.3658284369929397
			micro iteration [  4 ] ::      -2.3658292945820438
			micro iteration [  5 ] ::      -2.3658294215727129
			micro iteration [  6 ] ::      -2.3658294363946992
			--------------------------------------------------
		done (     0.03 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.03 sec. 
		--------------------------------------------------------------------------------
			Energy =      -2.3658294363946992
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -2.3658294363946988
			micro iteration [  2 ] ::      -2.7646505305754920
			micro iteration [  3 ] ::      -2.8107501264149075
			micro iteration [  4 ] ::      -3.2118637416570786
			micro iteration [  5 ] ::      -3.2401082008853974
			micro iteration [  6 ] ::      -3.2664186592480848
			micro iteration [  7 ] ::      -3.2923515320005512
			micro iteration [  8 ] ::      -3.2930883204513854
			micro iteration [  9 ] ::      -3.2930929453901916
			micro iteration [ 10 ] ::      -3.3333080125899524
			--------------------------------------------------
			macro iteration [  1 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3333080125899515
			micro iteration [  2 ] ::      -3.3408197949052050
			micro iteration [  3 ] ::      -3.3408443253771316
			micro iteration [  4 ] ::      -3.3408446136893137
			micro iteration [  5 ] ::      -3.3408450886892171
			micro iteration [  6 ] ::      -3.3408456195169229
			micro iteration [  7 ] ::      -3.3408460418340700
			micro iteration [  8 ] ::      -3.3408475537184645
			micro iteration [  9 ] ::      -3.3408475728056004
			micro iteration [ 10 ] ::      -3.3408476172483477
			--------------------------------------------------
		done (     0.08 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.09 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483477
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			BACKWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   3 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483454
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483454
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.01 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483468
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483468
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
	====================================================================================================
		SWEEP ITERATION [    0 ] :: ROOT =  0 FINISHED
			Sweep Energy =      -3.3408476172483477 ( delta E = -3.34e+00 ) 

	====================================================================================================
		SWEEP ITERATION [    1 ] :: ROOT =  0
	====================================================================================================
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			FORWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   0 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483463
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483463
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483446
			--------------------------------------------------
		done (     0.01 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483446
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483454
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483454
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			BACKWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   3 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483450
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483450
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483454
			--------------------------------------------------
		done (     0.01 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483454
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 2-site algorithm 
		computing 2-site wavefunction...
		done (     0.00 sec. ) 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
	====================================================================================================
		SWEEP ITERATION [    1 ] :: ROOT =  0 FINISHED
			Sweep Energy =      -3.3408476172483463 ( delta E = 1.33e-15 ) 

	====================================================================================================
		SWEEP ITERATION [    0 ] :: ROOT =  0
	====================================================================================================
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			FORWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   0 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483463
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483463
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483463
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483463
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			BACKWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   3 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483454
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483454
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483454
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483454
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
	====================================================================================================
		SWEEP ITERATION [    0 ] :: ROOT =  0 FINISHED
			Sweep Energy =      -3.3408476172483463 ( delta E = 0.00e+00 ) 

	====================================================================================================
		SWEEP ITERATION [    1 ] :: ROOT =  0
	====================================================================================================
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			FORWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   0 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483454
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483454
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction of this site (sys)
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
			BACKWARD SWEEP
	++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
	====================================================================================================
		SITE [   3 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483450
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483450
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   2 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483459
			--------------------------------------------------
		done (     0.01 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.01 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483459
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
		SITE [   1 ] 
	----------------------------------------------------------------------------------------------------
		loading operators and wavefunction of next site (env)
		optimizing wavefunction: 1-site algorithm 
		computing diagonal elements...
		done (     0.00 sec. ) 
		optimizing wavefunction (Davidson solver)...
			macro iteration [  0 ] 
			--------------------------------------------------
			micro iteration [  1 ] ::      -3.3408476172483454
			--------------------------------------------------
		done (     0.00 sec. ) 
		doing singular value decomposition on wavefunction...
		done (     0.00 sec. ) 
		renormalizing operators to the next...
		done (     0.00 sec. ) 
		--------------------------------------------------------------------------------
		Total time for optimization:     0.00 sec. 
		--------------------------------------------------------------------------------
			Energy =      -3.3408476172483454
		--------------------------------------------------------------------------------
		saving operators and wavefunction for this site (sys)
	====================================================================================================
	====================================================================================================
		SWEEP ITERATION [    1 ] :: ROOT =  0 FINISHED
			Sweep Energy =      -3.3408476172483459 ( delta E = 4.44e-16 ) 

	====================================================================================================
			Sweep Energy =      -3.3408476172483459

